driver-core/src/main/com/mongodb/internal/async/AsyncSupplier.java

```diff
@@ -81,6 +81,26 @@ default void finish(final SingleResultCallback<T> callback) {
         }
     }
 
+    /**
+     * The runnable will always be executed, including on the exceptional path.
+     * @param runnable the runnable
+     * @param callback the callback
+     */
+    default void thenAlwaysRunAndFinish(final Runnable runnable, final SingleResultCallback<T> callback) {
```

Comment:
`AsyncRunnable` is `AsyncSupplier<Void>`, and currently has the method `thenAlwaysRunAndFinish(Runnable runnable, SingleResultCallback<Void> callback)`, which now overrides the new `thenAlwaysRunAndFinish` method from `AsyncSupplier` for no reason. Let's just uplift `AsyncRunnable.thenRunAndFinish`, `thenAlwaysRunAndFinish` from `AsyncRunnable` to `AsyncSupplier`, without changing either the implementations or the docs.